tasks/bond_configuration.yml

```diff
@@ -79,6 +79,7 @@
   with_subelements:
     - "{{ interfaces_bond_interfaces }}"
     - bond_slaves
+  when: item.0.device not in interfaces_bridge_interfaces | selectattr("ports", "defined") | map(attribute='ports') | flatten
```

Comment:
Why would we not want to define config for slaves, if the master is plugged into a bridge?

Reply:
If the config for slaves are installed, I have the error messages below with Debian 11:
```
Aug 19 10:59:01 server systemd[1]: Starting Raise network interfaces...
Aug 19 10:59:01 server ifup[34668]: No iface stanza found for master bond0
Aug 19 10:59:01 server ifup[34665]: run-parts: /etc/network/if-pre-up.d/ifenslave exited with return code 1
Aug 19 10:59:01 server ifup[34607]: ifup: failed to bring up eno5
Aug 19 10:59:01 server ifup[34674]: No iface stanza found for master bond0
Aug 19 10:59:01 server ifup[34671]: run-parts: /etc/network/if-pre-up.d/ifenslave exited with return code 1
Aug 19 10:59:01 server ifup[34607]: ifup: failed to bring up ens1f0
Aug 19 10:59:01 server systemd[1]: networking.service: Main process exited, code=exited, status=1/FAILURE
Aug 19 10:59:01 server systemd[1]: networking.service: Failed with result 'exit-code'.
Aug 19 10:59:01 server systemd[1]: Failed to start Raise network interfaces.
```

Reply:
Looks like the bond master config is missing? Is that because it gets overwritten by the bridge port config?

Reply:
In fact, even without bridge, it does not work ...

If i use this config:
```yaml
interfaces_bond_interfaces:
  - device: bond1                                                                                   
    bootproto: static                                                                               
    bond_mode: 802.3ad                                                                              
    bond_miimon: 100                                                                                
    bond_lacp_rate: 1                                                                               
    bond_min_links: 1                                                                               
    bond_xmit_hash_policy: layer3+4                                                                 
    bond_slaves: [eno6, ens1f1]                                                                     
    address: 192.168.56.130                                                                           
    netmask: 255.255.255.0
```

I have the error message below;
```
  rc: 1
  start: '2022-08-19 12:14:26.623099'
  stderr: |-
    nohup: ignoring input
    ifdown: interface bond1 not configured
    ifdown: interface ens1f1 not configured
    ifdown: interface eno6 not configured
    No iface stanza found for master bond1
    run-parts: /etc/network/if-pre-up.d/ifenslave exited with return code 1
    ifup: failed to bring up eno6
    No iface stanza found for master bond1
    run-parts: /etc/network/if-pre-up.d/ifenslave exited with return code 1
    ifup: failed to bring up ens1f1
  stderr_lines: <omitted>
  stdout: |-
    Failed to bring up interface eno6
    Failed to bring up interface ens1f1
  stdout_lines: <omitted>
```

Perhaps the behavior is different on Debian 11 ...

Reply:
Here is the config of eno6:
```
# Ansible managed

auto eno6
iface eno6 inet manual
bond-master bond1
```

The bond-master config is OK ?

Reply:
Check the config for bond1

Reply:
Here is the config of bond1:
```
# Ansible managed

auto bond1
iface bond1 inet static
address 192.168.56.130
netmask 255.255.255.0
bond-mode 802.3ad
bond-miimon 100
bond-lacp-rate 1
bond-min-links 1
bond-xmit_hash_policy layer3+4
bond-slaves eno6 ens1f1
```

Reply:
Perhaps related to https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=990428 ?

Reply:
Do you agree if I add a boolean option instead ?

Reply:
I don't personally use this role on debian/ubuntu, so I don't feel that comfortable recommending any particular solution. We do need to ensure it works on Debian and Ubuntu though.

Reply:
I add the option interfaces_bond_setup_slaves which is true by default to keep the backward compatibility.